contracts/ValidatorMetadata.sol

```diff
@@ -48,39 +24,134 @@ contract ValidatorMetadata {
+        lastName = bytes32Storage[keccak256("validators", _miningKey, "lastName")];
+        licenseId = bytes32Storage[keccak256("validators", _miningKey, "licenseId")];
+        fullAddress = stringStorage[keccak256("validators", _miningKey, "fullAddress")];
+        state = bytes32Storage[keccak256("validators", _miningKey, "state")];
+        zipcode = uintStorage[keccak256("validators", _miningKey, "zipcode")];
+        expirationDate = uintStorage[keccak256("validators", _miningKey, "expirationDate")];
+        createdDate = uintStorage[keccak256("validators", _miningKey, "createdDate")];
+        updatedDate = uintStorage[keccak256("validators", _miningKey, "updatedDate")];
+        minThreshold = uintStorage[keccak256("validators", _miningKey, "minThreshold")];
+    }
+
+    function pendingChanges(address _miningKey) public view returns (
+        bytes32 firstName,
+        bytes32 lastName,
+        bytes32 licenseId,
+        string fullAddress,
+        bytes32 state,
+        uint256 zipcode,
+        uint256 expirationDate,
+        uint256 createdDate,
+        uint256 updatedDate,
+        uint256 minThreshold
+    ) {
+        firstName = bytes32Storage[keccak256("pendingChanges", _miningKey, "firstName")];
+        lastName = bytes32Storage[keccak256("pendingChanges", _miningKey, "lastName")];
+        licenseId = bytes32Storage[keccak256("pendingChanges", _miningKey, "licenseId")];
+        fullAddress = stringStorage[keccak256("pendingChanges", _miningKey, "fullAddress")];
+        state = bytes32Storage[keccak256("pendingChanges", _miningKey, "state")];
+        zipcode = uintStorage[keccak256("pendingChanges", _miningKey, "zipcode")];
+        expirationDate = uintStorage[keccak256("pendingChanges", _miningKey, "expirationDate")];
+        createdDate = uintStorage[keccak256("pendingChanges", _miningKey, "createdDate")];
+        updatedDate = uintStorage[keccak256("pendingChanges", _miningKey, "updatedDate")];
+        minThreshold = uintStorage[keccak256("pendingChanges", _miningKey, "minThreshold")];
+    }
+
+    function confirmations(address _miningKey) public view returns (
+        uint256 count,
+        address[] voters
+    ) {
+        return (
+            uintStorage[keccak256("confirmations", _miningKey, "count")],
+            addressArrayStorage[keccak256("confirmations", _miningKey, "voters")]
+        );
+    }
+
+    function pendingProxyConfirmations(address _newProxyAddress) public view returns (
+        uint256 count,
+        address[] voters
+    ) {
+        bytes32 countHash = keccak256("pendingProxyConfirmations", _newProxyAddress, "count");
+        bytes32 votersHash = keccak256("pendingProxyConfirmations", _newProxyAddress, "voters");
+
+        return (
+            uintStorage[countHash],
+            addressArrayStorage[votersHash]
+        );
+    }
+
+    function initProxyAddress(address _proxyStorage) public {
```

Comment:
There is no restriction, who can call this method. Let's limit to the creator of contract.